finder-app/writer.c

```diff
@@ -0,0 +1,61 @@
+#include <fcntl.h>
+#include <syslog.h>
+#include <errno.h>
+#include <string.h>
+#include <stdlib.h>
+#include <unistd.h>
+
+#define DEBUG(fmt, ...) do{syslog(LOG_USER|LOG_DEBUG, fmt __VA_OPT__(,) __VA_ARGS__);} while (0)
+#define ERROR(fmt, ...) do { \
+	syslog(LOG_USER|LOG_ERR, fmt __VA_OPT__(,) __VA_ARGS__); \
+	exit(1); \
```

Comment:
@georgehodgkins I like the macro here but the inline exit is dangerous for missing cleanup, for instance file descriptor closes below.